File: tests/benchmark/vulnerable/go/cmdi_001.go
// BENCHMARK: vulnerable - cmdi
package main

import (
	"fmt"
	"net/http"
	"os/exec"
)

func runHandler(w http.ResponseWriter, r *http.Request) {
	userInput := r.URL.Query().Get("cmd")
	out, err := exec.Command("sh", "-c", userInput).Output()
	if err != nil {
		http.Error(w, err.Error(), 500)
		return
	}
	fmt.Fprintf(w, "%s", out)
}

func main() {
	http.HandleFunc("/run", runHandler)
	http.ListenAndServe(":8080", nil)
}

File: tests/benchmark/vulnerable/javascript/cmdi_001.js
// BENCHMARK: vulnerable - cmdi
const express = require('express');
const { exec } = require('child_process');
const app = express();

app.get('/run', (req, res) => {
    const cmd = req.query.cmd;
    exec('ls -la ' + cmd, (error, stdout, stderr) => {
        res.send(stdout);
    });
});

app.listen(3000);
